File: Taverne/src/app/model/inventaire/bar.ts
import { Intervention } from './../fonctionnalitees/intervention';
import { Events } from './../fonctionnalitees/events';
import { Employe } from './../comptes/employe';
import { Stock } from './stock';
export class Bar {
  constructor(
    private _idBar?: number,
    private _nom?: string,
    private _stocks?: Stock,
    private _events?: Events,
    private _employes?: Employe,
    private _interventions?: Intervention
  ) {}

  public get idBar(): number | undefined {
    return this._idBar;
  }

  public set idBar(value: number | undefined) {
    this._idBar = value;
  }

  public get nom(): string | undefined {
    return this._nom;
  }

  public set nom(value: string | undefined) {
    this._nom = value;
  }

  public get stocks(): Stock | undefined {
    return this._stocks;
  }

  public set stocks(value: Stock | undefined) {
    this._stocks = value;
  }

  public get events(): Events | undefined {
    return this._events;
  }

  public set events(value: Events | undefined) {
    this._events = value;
  }

  public get employes(): Employe | undefined {
    return this._employes;
  }

  public set employes(value: Employe | undefined) {
    this._employes = value;
  }

  public get interventions(): Intervention | undefined {
    return this._interventions;
  }

  public set interventions(value: Intervention | undefined) {
    this._interventions = value;
  }
}

File: TaverneSpringBoot/src/main/java/soprajc/TaverneSpringBoot/model/inventaire/Bar.java
package soprajc.TaverneSpringBoot.model.inventaire;

import java.util.Objects;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonView;

import soprajc.TaverneSpringBoot.model.JsonViews;
import soprajc.TaverneSpringBoot.model.comptes.Employe;
import soprajc.TaverneSpringBoot.model.fonctionnalitees.Events;
import soprajc.TaverneSpringBoot.model.fonctionnalitees.Intervention;

@Entity
public class Bar {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id_bar")
	@JsonView(JsonViews.Evenement.class)
	private Long idBar;
	@JsonView(JsonViews.Evenement.class)
	private String nom;

	@OneToMany(mappedBy = "bar")
	Set<Stock> stocks;

	@OneToMany(mappedBy = "bar")
	Set<Events> events;

	@OneToMany(mappedBy = "bar")
	Set<Employe> employes;

	/////
	@OneToMany(mappedBy = "bar")
	Set<Intervention> interventions;
	////

	public Bar() {
	}
	
	public Bar(Long idBar, String nom) {
		this.idBar=idBar;
		this.nom=nom;
	}
	
	
	public Bar(String nom, Set<Stock> stocks, Set<Events> events, Set<Employe> employes,
			Set<Intervention> interventions) {
		this.nom = nom;
		this.stocks = stocks;
		this.events = events;
		this.employes = employes;
		this.interventions = interventions;
	}

	public Long getIdBar() {
		return idBar;
	}

	public void setIdBar(Long idBar) {
		this.idBar = idBar;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public Set<Stock> getStocks() {
		return stocks;
	}

	public void setStocks(Set<Stock> stocks) {
		this.stocks = stocks;
	}

	public Set<Events> getEvents() {
		return events;
	}

	public void setEvents(Set<Events> events) {
		this.events = events;
	}

	public Set<Employe> getEmployes() {
		return employes;
	}

	public void setEmployes(Set<Employe> employes) {
		this.employes = employes;
	}

	public Set<Intervention> getInterventions() {
		return interventions;
	}

	public void setInterventions(Set<Intervention> interventions) {
		this.interventions = interventions;
	}

	@Override
	public int hashCode() {
		return Objects.hash(idBar);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Bar other = (Bar) obj;
		return Objects.equals(idBar, other.idBar);
	}

}
